pkg/scalers/temporal.go

```diff
@@ -0,0 +1,243 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	sdk "go.temporal.io/sdk/client"
+	sdklog "go.temporal.io/sdk/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+var (
+	temporalDefauleQueueTypes = []sdk.TaskQueueType{
+		sdk.TaskQueueTypeActivity,
+		sdk.TaskQueueTypeWorkflow,
+		sdk.TaskQueueTypeNexus,
+	}
+)
+
+type temporalScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *temporalMetadata
+	tcl        sdk.Client
+	logger     logr.Logger
+}
+
+type temporalMetadata struct {
+	Endpoint                  string   `keda:"name=endpoint,                  order=triggerMetadata;resolvedEnv"`
+	Namespace                 string   `keda:"name=namespace,                 order=triggerMetadata;resolvedEnv, default=default"`
+	ActivationTargetQueueSize int64    `keda:"name=activationTargetQueueSize, order=triggerMetadata, default=0"`
+	TargetQueueSize           int64    `keda:"name=targetQueueSize,           order=triggerMetadata, default=5"`
+	QueueName                 string   `keda:"name=queueName,                 order=triggerMetadata;resolvedEnv"`
+	QueueTypes                []string `keda:"name=queueTypes,                order=triggerMetadata, optional"`
+	BuildIDs                  []string `keda:"name=buildIds,                  order=triggerMetadata;resolvedEnv, optional"`
+	AllActive                 bool     `keda:"name=selectAllActive,           order=triggerMetadata, default=true"`
+	Unversioned               bool     `keda:"name=selectUnversioned,         order=triggerMetadata, default=true"`
+	APIKey                    string   `keda:"name=apiKey,                    order=authParams;resolvedEnv;triggerMetadata, optional"`
+
+	UnsafeSsl   bool   `keda:"name=unsafeSsl,                 order=triggerMetadata, optional"`
+	Cert        string `keda:"name=cert,                      order=authParams;resolvedEnv, optional"`
```

Comment:
I think it's more standard in Temporal deployments to have env vars that point to files for the key/cert rather than have them present in the environment directly. It would be good to have CertPath/KeyPath/CaPath or similar.

Reply:
@robholland That's currently not possible with KEDA. KEDA can only read from environment variables or secrets. Ideally, in k8s, certificates and keys are stored in secrets, so this can be easily managed using TriggerAuthentication.

Reply:
The file paths are normally pointed to certs that are mounted into the pods from secrets, so presumably users can just point at the secrets directly for KEDA use.

Reply:
Yes it can be done with `TriggerAuthentication`.

Reply:
Our mTLS client certificates and keys auto rotate, live on the filesystem, at least every hour. Our worker code handles the rotation without restarting.

I think any code using mTLS to connect to Temporal needs to be able to handle certificates and keys that are rotated frequently. Getting the certificates and keys from the filesystem seems natural and normal, if there is a way for KEDA to support that it would be good.

We use a sidecar to write/rotate the certificates and keys. But there are other Kubernetes systems for mTLS certificate management that also use filesystems and not Kubernetes Secret resources, like [cert-manager csi-driver-spiffe](https://cert-manager.io/docs/usage/csi-driver-spiffe/).